### PR TITLE
Replace home ID with config entry title in Z-Wave device Info

### PR DESCRIPTION
**Pull request by cgarwood**
## Proposed change

Replace the Home ID in the Z-Wave device info with the config entry title. Home ID is more advanced and less useful for an average user to differentiate multiple Z-Wave JS networks.

image

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue or discussion:
- Link to documentation pull request:

## Checklist

- [x] The code change is tested and works locally.
- [x] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

File: src/panels/config/devices/device-detail/integration-elements/zwave_js/ha-device-info-zwave_js.ts [hunk 1]
```diff
@@ -52,20 +58,26 @@ export class HaDeviceInfoZWaveJS extends LitElement {
...
     }
     return html`
       <h4>
         ${this.hass.localize("ui.panel.config.zwave_js.device_info.zwave_info")}
       </h4>
       <div>
-        ${this.hass.localize("ui.panel.config.zwave_js.common.home_id")}:
-        ${this._homeId}
+        ${this.hass.localize("ui.panel.config.zwave_js.common.source")}:
+        ${this._configEntry.title}
```
Comment (bramkragten) [inline src/panels/config/devices/device-detail/integration-elements/zwave_js/ha-device-info-zwave_js.ts:80]: Should we even show this if there is only 1 zwave js config entry?

Comment (cgarwood) [reply src/panels/config/devices/device-detail/integration-elements/zwave_js/ha-device-info-zwave_js.ts:80]: Ooo good idea. Definitely not necessary if there's only 1 instance of zwavejs set up

File: src/panels/config/devices/device-detail/integration-elements/zwave_js/ha-device-info-zwave_js.ts [hunk 2]
```diff
@@ -52,21 +57,36 @@ export class HaDeviceInfoZWaveJS extends LitElement {
...
     }
+
+    const configEntries = await getConfigEntries(this.hass);
+    this._configEntry = configEntries.find(
+      (entry) => entry.entry_id === this._entryId!
+    );
+
+    const zwaveJSIntegrations = configEntries.filter(
+      (entry) => entry.domain === "zwave_js"
+    );
```
Comment (bramkragten) [inline src/panels/config/devices/device-detail/integration-elements/zwave_js/ha-device-info-zwave_js.ts:68]: Let's filter first and then search in those results for the config entry :-)

Or do it all in 1 loop:

```ts
let zwaveJsConfEntries = 0;
for (const entry of configEntries) {
	if (entry.domain !== "zwave_js") {
		continue;
	}
	if (zwaveJsConfEntries) {
		this._multipleConfigEntries = true;
	}
	if (entry.entry_id === this._entryId!) {
		this._configEntry = entry;
	}
	if (this._configEntry && this._multipleConfigEntries) {
		break;
	}
}
```

File: src/panels/config/devices/device-detail/integration-elements/zwave_js/ha-device-info-zwave_js.ts [hunk 3]
```diff
@@ -52,21 +57,44 @@ export class HaDeviceInfoZWaveJS extends LitElement {
...
+      }
+      zwaveJsConfEntries++;
+    }
+
     this._node = await fetchNodeStatus(this.hass, this._entryId, this._nodeId);
   }
 
   protected render(): TemplateResult {
-    if (!this._node) {
+    if (!this._node || !this._configEntry) {
```
Comment (bramkragten) [inline src/panels/config/devices/device-detail/integration-elements/zwave_js/ha-device-info-zwave_js.ts:83]:
```suggestion
    if (!this._node) {
```

**Review by bramkragten — DISMISSED** [review]

**Review by bramkragten — APPROVED** [review]

**Pull request merged**